lib/Parser.js

```diff
@@ -37,6 +38,107 @@ const POSSIBLE_AST_OPTIONS = [{
 	}
 }];
 
+class StackedSetMap {
+	constructor(defaultValue, parentStack) {
+		this.defaultValue = defaultValue;
+		this.stack = parentStack === undefined ? [] : parentStack.slice();
+		this.map = new Map();
+		this.stack.push(this.map);
+	}
+
+	add(item) {
+		this.map.set(item, true);
+	}
+
+	set(item, value) {
+		this.map.set(item, value);
+	}
+
+	delete(item) {
+		this.map.set(item, false);
+	}
+
+	has(item) {
+		return this.get(item, false);
```

Comment:
This passes a second value to get, but it only accepts / uses one.

This also has an inconsistent return type (anything) instead of boolean.